src/client/datascience/interactive-ipynb/digestStorage.ts

```diff
@@ -102,15 +102,16 @@ export class DigestStorage implements IDigestStorage {
             }
 
             // If it doesn't exist, create one
-            const randomBytesBuffer = await this.prng.randomBytes(1024);
-            const key = randomBytesBuffer.toString('hex');
+            let key = await this.prng.generateRandomKey(1024);
             if (key) {
-                traceInfo(`Successfully generated keyfile`);
-                await this.fs.writeLocalFile(defaultKeyFileLocation, key);
-                return key;
+                traceInfo(`Successfully generated keyfile with OS PRNG`);
             } else {
-                traceError('Failed to initialize secret for notebook trust');
+                // Fallback to crypto.randomBytes
+                traceError('Failed to initialize secret for notebook trust with OS PRNG');
+                key = randomBytes(1024).toString('hex');
```

Comment:
I think we should capture the telemetry here to see how often this fails.
If we don't care, then why not always use `randomBytes`

Reply:
I put the telemetry capture in randomBytes itself. `Telemetry.NativeRandomBytesGenerationFailed`